src/components/modals/SupportRequestModal.js

```diff
@@ -0,0 +1,194 @@
+import { SupportRequestBaseModal } from '../SupportRequestBaseModal';
+import { Alert } from '../Alert';
+import { Support, User } from '../../libs/ajax';
+import { Storage } from "../../libs/storage";
+import { convertCompilerOptionsFromJson } from 'typescript';
+
+export const SupportRequestModal = hh(class SupportRequestModal extends Component {
+  constructor(props) {
+    super(props);
+    const isLogged = Storage.userIsLogged();
+    let name = isLogged ? Storage.getCurrentUser().displayName : '';
+    let first_name = isLogged ? ", " + name.split(" ")[0] : '';
+    let email = isLogged ? Storage.getCurrentUser().email : '';
+    let height = isLogged ? "550px": '700px';
+    let top = isLogged ? '400px': '1';
+    this.state = {
+      name: name,
+      isLogged: isLogged,
+      type: 'question',
+      subject: '',
+      description: '',
+      attachment: '',
+      email: email,
+      first_name: first_name,
+      height: height,
+      top: top,
+      vaild: false
+    };
+    this.closeHandler = this.closeHandler.bind(this);
+    this.afterOpenHandler = this.afterOpenHandler.bind(this);
+    this.OKHandler = this.OKHandler.bind(this);
+    this.nameChangeHandler = this.nameChangeHandler.bind(this);
+    this.typeChangeHandler = this.typeChangeHandler.bind(this);
+    this.subjectChangeHandler = this.subjectChangeHandler.bind(this);
+    this.descriptionChangeHandler = this.descriptionChangeHandler.bind(this);
+    this.attachmentChangeHandler = this.attachmentChangeHandler.bind(this);
+    this.emailChangeHandler = this.emailChangeHandler.bind(this);
+  }
+
+  async OKHandler() {
+    const ticket = {};
+    ticket.request = {
+      requester: { name: this.state.name, email: this.state.email },
+      subject: this.state.subject,
+      // BEWARE changing the following ids or values! If you change them then you must thoroughly test.
+      custom_fields: [
+        { id: 360012744452, value: this.state.type},
+        { id: 360007369412, value: this.state.description},
+        { id: 360012744292, value: this.state.name},
+        { id: 360012782111, value: this.state.email },
+        { id: 360018545031, value: this.state.email }
+      ],
+      comment: {
+        body: this.state.description + '\n\n------------------\nSubmitted from: ' + this.props.url,
+        //${currUrl}
+        uploads: [this.state.attachment.token]
+      },
+      ticket_form_id: 360000669472
+    };
+    await Support.createSupportRequest(ticket);
+    
+    await this.setState(prev => {
+      prev.name = '';
+      prev.type = "question";
+      prev.subject = '';
+      prev.description = '';
+      prev.attachment = '';
+      prev.email = '';
+      return prev;
+    });
+    this.props.onOKRequest('support');;
+  }
+
+  closeHandler() {
+    // this is the method to handle Cancel click
+    // could do some cleaning here
+    // or delegate it to the parent
+    // we need to use it to close the
+    // DO SOMETHING HERE ...
```

Comment:
You could add a little noty notification here, "Support request canceled" that goes away after a few seconds. Just a suggestion - I'm open to whatever you decide.